### PR TITLE
Add Basic cluster tests for attributes constraints

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

There is no specific tests for the `Basic` cluster. 

#### Change overview
 * Move the `Basic` cluster definition out of the Silabs's definition and update it a little
 * Add `Test_10_1_1` for the `Basic cluster`
 * Update the templates to support `optional` keyword, such that an attribute is tested only if it is supported on the accessory side
 * Update the templates to support `constraints` keyword
 * Add `minLength`/`maxLength` constraints support (`format` and `type` still needs to be done)
 * Update gen/ folders

#### Testing
 * Tests were generated for the `Darwin CI` and the test suites using `chip-tool`

### CODE REVIEW

**Review by msandstedt — APPROVED** [review]

File: src/app/zap-templates/zcl/basic-information-cluster.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    <attribute side="server" code="0x000A" define="SOFTWARE_VERSION_STRING"   type="CHAR_STRING"             length="64"  writable="false" optional="false">SoftwareVersionString</attribute>
+    <attribute side="server" code="0x000B" define="MANUFACTURING_DATE"        type="CHAR_STRING"             length="16"  writable="false" optional="true" >ManufacturingDate</attribute>
+    <attribute side="server" code="0x000C" define="PART_NUMBER"               type="CHAR_STRING"             length="254" writable="false" optional="true" >PartNumber</attribute><!-- TODO The actual length is not defined in the spec. But the current implementation expect a length -->
+    <attribute side="server" code="0x000D" define="PRODUCT_URL"               type="CHAR_STRING"             length="256" writable="false" optional="true" >ProductURL</attribute>
+    <attribute side="server" code="0x000E" define="PRODUCT_LABEL"             type="CHAR_STRING"             length="64"  writable="false" optional="true" >ProductLabel</attribute>
```
Comment (tcarmelveilleux) [inline src/app/zap-templates/zcl/basic-information-cluster.xml:41]: See spec for lengths: the value for PART_NUMBER is now available.

```
| 12 | PartNumber              | <<string-def>>  | max 32     | F       | MS      | R V      | O
| 13 | ProductURL              | <<string-def>>  | max 256    | F       | MS      | R V      | O
| 14 | ProductLabel            | <<string-def>>  | max 64     | F       | MS      | R V      | O
| 15 | SerialNumber            | <<string-def>>  | max 32     | F       | MS      | R V      | O
```

File: src/controller/data_model/controller-clusters.zap [hunk 2]
```diff
@@ -1232,7 +1232,7 @@
               "storageOption": "RAM",
```
Comment (tcarmelveilleux) [inline src/controller/data_model/controller-clusters.zap:1232]: Is there a way to avoid storageOption: RAM for all the fixed values?

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@Damian-Nordic @jepenven-silabs ?

File: src/controller/data_model/controller-clusters.zap (hunk 2 above)
Comment (bzbarsky-apple) [reply src/controller/data_model/controller-clusters.zap:1232]: https://github.com/project-chip/connectedhomeip/issues/5926 tracks that...

**Review by bzbarsky-apple — APPROVED** [review]
Looks good, but @tcarmelveilleux's comment about the max length there should be addressed, either here or in a followup.

File: src/app/zap-templates/zcl/basic-information-cluster.xml (hunk 1 above)
Comment (vivien-apple) [reply src/app/zap-templates/zcl/basic-information-cluster.xml]: I have also updated again the format of the `basic-information-cluster.xml`, in order to try to makes it easier to spot differences with the spec.

**Comment by woody-apple** [comment]
@vivien-apple conflicts :(

**Review by andy31415 — APPROVED** [review]
Needs conflict resolution.

Cannot easily review this due to  large amoung of generated code.

**Comment by bzbarsky-apple** [comment]
> Cannot easily review this due to large amoung of generated code.

@andy31415 fwiw, reviewing the individual changesets worked pretty well for me for this one....

**Pull request merged**